File: servers/scalatra/src/main/scala/apis/InstrumentApi.scala
package apis

import com.wordnik.client.model.Instrument
import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class InstrumentApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "InstrumentApi"
  override protected val applicationName: Option[String] = Some("instrument")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }



  val getOperation = (apiOperation[List[Instrument]]("get")
      summary "Get instruments."
      parameters(
        queryParam[Any]("filter").description("").optional)
  )

  get("/",operation(getOperation)) {
    val filter = params.getAs[Any]("filter")
    println("filter: " + filter)
  }




  val getActiveOperation = (apiOperation[List[Instrument]]("getActive")
      summary "Get all active instruments and instruments that have expired in <24hrs."
      parameters(
        )
  )

  get("/active",operation(getActiveOperation)) {
    }

}

File: servers/jax-rs/src/main/java/com/wordnik/api/InstrumentApi.java
package com.wordnik.api;

import com.wordnik.swagger.annotations.*;

import com.wordnik.client.model.Instrument;
import java.util.List;
import com.wordnik.api.NotFoundException;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Path("/instrument.json")
@Api(value = "/instrument", description = "the instrument API")
@Produces({"application/json"})
public class InstrumentApi {
  @GET
  @Path("/")
  @ApiOperation(value = "Get instruments.", notes = "", responseClass = "List<Instrument>")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response get(
    @ApiParam(value = "Table filter. For example, send {"symbol": "XBTF15"}."
    ,required=true
)@QueryParam("filter")
 Object filter
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  @GET
  @Path("/active")
  @ApiOperation(value = "Get all active instruments and instruments that have expired in <24hrs.", notes = "", responseClass = "List<Instrument>")
  @ApiErrors(value = { @ApiError(code = 200, reason = "Request was successful"),@ApiError(code = 400, reason = "Parameter Error"),@ApiError(code = 401, reason = "Unauthorized"),@ApiError(code = 404, reason = "Not Found")})
     
  public Response getActive(
    )
      throws NotFoundException {
      // do some magic!
      return Response.ok().entity(new ApiResponse(ApiResponse.OK, "magic!")).build();
  }

  }

